source/puppet/4.3/reference/deprecated_api.markdown

```diff
@@ -6,19 +6,27 @@ title: "Deprecated Extension Points and APIs"
 
 The following APIs and extension points are deprecated, and will be removed in Puppet 5.0.
 
+## Puppet Faces is a private API
 
-Resource Type and Provider APIs
------
+(This isn't actually a deprecation, since we don't have a specific plan to remove the Faces APIs. But we needed a place to mention this on the record, and the deprecations section seemed like the best long-term bet.)
 
+The `Puppet::Face` and `Puppet::Indirector::Face` classes are private APIs, and not particularly good ones. Please don't use them. If you want to develop CLI applications that use Puppet's data, you should:
+
+* Pick a popular, well-maintained Ruby framework for CLI apps, like Thor or Commander.
+* Use Puppet's public Ruby APIs to access its settings and data. See the [developer documentation](./yard/frames.html) for comprehensive info about the classes and methods available for this.
+* Distribute your application as a Gem or native package, not as a Puppet module.
+* If you _really_ want your command to be available as a `puppet <NAME>`-style subcommand, you can name it `puppet-<NAME>` --- there's a secret feature that lets you leave out the hyphen.
```

Comment:
"and have your gem or package install it in $PATH"?  (or does it have to be in a special directory?)